lib/commands/kubernetesUndeploy.ts

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright © 2018 Atomist, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+    HandlerResult,
+    logger,
+    Parameter,
+    Parameters,
+} from "@atomist/automation-client";
+import {
+    CommandHandlerRegistration,
+    slackSuccessMessage,
+} from "@atomist/sdm";
+import { deleteApplication } from "../kubernetes/application";
+import { defaultNamespace } from "../kubernetes/namespace";
+
+@Parameters()
+export class KubernetesUndeployParameters {
+
+    @Parameter({
+        displayName: "Name",
+        description: "name of resources to remove",
+        pattern: /^[a-z](?:[-a-z0-9]*[a-z0-9])?$/,
+        validInput: "a valid Kubernetes resource name, beginning with a lowercase letter, ending with a lowercase" +
+            "letter or number, and containing only lowercase letters, numbers, and dashes (-)",
+        minLength: 1,
+        maxLength: 63,
+        required: true,
+    })
+    public name: string;
+
+    @Parameter({
+        displayName: "Namespace",
+        description: "namespace of resources to remove",
+        pattern: /^[a-z](?:[-a-z0-9]*[a-z0-9])?$/,
+        validInput: "a valid Kubernetes namespace, beginning with a lowercase letter, ending with a lowercase" +
+            "letter or number, and containing only lowercase letters, numbers, and dashes (-)",
+        minLength: 1,
+        maxLength: 63,
+        required: false,
+    })
+    public ns: string = defaultNamespace;
+
+}
+
+/**
+ * Safely remove all resources related to an Kubernetes deployment.
+ */
+export const kubernetesUndeploy: CommandHandlerRegistration<KubernetesUndeployParameters> = {
+    name: "KubernetesUndeploy",
+    intent: "kube undeploy",
+    description: "remove all resources related to an application from Kubernetes cluster",
+    paramsMaker: KubernetesUndeployParameters,
```

Comment:
Since we deprecated `paramsMaker` seems like we should move this to the new declarative style too.

Reply:
What does that look like?